### PR TITLE
remove the markdown related switches

### PR DESCRIPTION
**Pull request by thewilsonator**
This has been enabled by default for a couple of years now.

Might also need a spec PR, e: there doesn't seen to be any need for this. The spec only mentions the features.

https://forum.dlang.org/post/bcwupfeuaghvjzyunbng@forum.dlang.org

### CODE REVIEW

File: src/dmd/cli.d [hunk 1]
```diff
@@ -829,7 +827,6 @@ dmd -cov -unittest myprog.d
     static immutable reverts = [
         Feature("dip25", "useDIP25", "revert DIP25 changes https://github.com/dlang/DIPs/blob/master/DIPs/archive/DIP25.md"),
         Feature("intpromote", "fix16997", "revert integral promotions for unary + - ~ operators"),
-        Feature("markdown", "markdown", "disable Markdown replacements in Ddoc"),
```
Comment (ibuclaw) [inline src/dmd/cli.d:832]: Should reverts be deprecated first?

File: src/dmd/cli.d [hunk 2]
```diff
@@ -819,8 +819,6 @@ dmd -cov -unittest myprog.d
             "give deprecation messages about all usages of complex or imaginary types", true, true),
         Feature("tls", "vtls",
             "list all variables going into thread local storage"),
-        Feature("vmarkdown", "vmarkdown",
```
Comment (ibuclaw) [inline src/dmd/cli.d:822]: Should transitions be deprecated first?

File: src/dmd/cli.d (hunk 1 above)
Comment (thewilsonator) [reply src/dmd/cli.d:832]: Dunno. Part of this PR was to figure out what, if anything, breaks. Buildkite is green so probably not much. 
If this were a switch that affected codegen I'd be much more cautious, but this affects documentation only, and it's been enabled by default for several _years_ now.

File: src/dmd/cli.d (hunk 2 above)
Comment (thewilsonator) [reply src/dmd/cli.d:822]: ditto

**Comment by thewilsonator** [comment]
Hmm, I wonder if [this comment](https://github.com/dlang/dmd/pull/11654#discussion_r479789424) is still accurate?

**Comment by ibuclaw** [comment]
> Hmm, I wonder if [this comment](https://github.com/dlang/dmd/pull/11654#discussion_r479789424) is still accurate?

I assume the order would have been:
1. Undocument the preview in dmd
2. Remove the preview from the phobos makefiles
3. Deprecate the preview.

1 and 2 are interchangeable (and in-fact if 2 is merged first, then 1+3 can be done together).

**Comment by thewilsonator** [comment]
> Remove the preview from the phobos makefiles

It appears to already have been removed.

**Comment by thewilsonator** [comment]
Added changelog entry.

@WalterBright seems to be in favour of this: https://forum.dlang.org/post/t4mhhc$1blq$1@digitalmars.com

File: changelog/markdown.dd [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Rmoved the `-transition=markdown` and `-revert=markdown` switches
```
Comment (RazvanN7) [inline changelog/markdown.dd:1]:
```suggestion
Removed the `-transition=markdown` and `-revert=markdown` switches
```

**Review by RazvanN7 — APPROVED** [review]
I think this is fine. I find it hard to imagine that anyone goes through the pain of using `-revert=markdown`

File: changelog/markdown.dd (hunk 3 above)
Comment (thewilsonator) [reply changelog/markdown.dd]: Fixed

**Pull request merged**